### DISCUSSION TITLE
STM32L4 host implementation

### DISCUSSION THREAD

**Discussion by friesendrywall** [hathach/tinyusb#1315] [General]
Is there any documentation with an overview of porting an MCU as a host?  Any idea how much time this would take?

**Comment by Ryzee119** [comment]
I have a stm32f7 series host port on my personal projects using HAL_HCD. Probably a few subtle differences but may be a start
https://gist.github.com/Ryzee119/5a1b74e98449fdf5ed24b3f361ff4e2c

You would also need to enable to usb gpio, clocks and usb IRQ etc not shown here
